Add recipes for Jackson 3.x exception changes

wakingrufus opened the pull request:
## What's changed?
Make changes to exception usage according to point 7 of https://github.com/FasterXML/jackson/wiki/Jackson-Release-3.0#major-changesfeatures-in-30 with additional details in https://github.com/FasterXML/jackson-future-ideas/wiki/JSTEP-4


## What's your motivation?
increase coverage of jackson 2->3 migration


### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

timtebeek commented:
What's possibly not helping here is the missing type on the import; although I'm not sure if that's really the issue.
![image](https://github.com/openrewrite/rewrite-jackson/assets/1027334/219c6e59-b3db-4bed-9366-86e458ae52d0)

wakingrufus committed:
Add recipes for Jackson 3.x exception changes

Make changes to exception usage according to point 7 of https://github.com/FasterXML/jackson/wiki/Jackson-Release-3.0#major-changesfeatures-in-30 with additional details in https://github.com/FasterXML/jackson-future-ideas/wiki/JSTEP-4

wakingrufus commented on src/test/java/org/openrewrite/jackson/UpgradeJackson_2_3Test.java:
```diff
@@ -48,46 +48,46 @@ void jacksonUpgradeToVersion3() {
         rewriteRun(
           //language=xml
           pomXml(
-              """
```
formatter fixed the indentation
